### ISSUE TITLE
WSL is not installing

### ISSUE THREAD

**Issue by roaringsundew40** [microsoft/WSL#5833]
I downloaded the Ubuntu 20.04 LTS and I ran it. This is the output it gave me.
`Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x8007019e
The Windows Subsystem for Linux optional component is not enabled. Please enable it and try again.
See https://aka.ms/wslinstall for details.
Press any key to continue...`
I looked it up online I found that I have to use the command 
`dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart`. 
But that gives the error 
`Deployment Image Servicing and Management tool
Version: 10.0.17134.1

Image Version: 10.0.17134.765

Error: 0x800f0831

Feature name Microsoft-Windows-Subsystem-Linux is unknown.
A Windows feature name was not recognized.
Use the /Get-Features option to find the name of the feature in the image and try the command again.

The DISM log file can be found at C:\Windows\Logs\DISM\dism.log`

I also found out that that I can use the GUI on Turn Windows Features on or off. But mine does not seem to load. 

![image](https://user-images.githubusercontent.com/58274772/91669350-1df2d100-eac9-11ea-9df7-352931f26c52.png)

I found this [article](https://teckangaroo.com/fix-wsl-components-not-enabled-error-windows-10/)

It says to use the command 
`Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux`
 But that gives the error
`Enable-WindowsOptionalFeature : Feature name Microsoft-Windows-Subsystem-Linux is unknown.
At line:1 char:1
+ Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows- ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Enable-WindowsOptionalFeature], COMException
    + FullyQualifiedErrorId : Microsoft.Dism.Commands.EnableWindowsOptionalFeatureCommand`

So, in summary, I am trying to run Ubuntu 20.04 LTS but it is telling me to install WSL Components but anything I do to install is not working. Could anyone help me with this?

Thanks!

**Comment by therealkenc** [comment]
> 10.0.17134.765

That is going to be too far back for a repro. If you can't get a list of features in "Turn Windows features on or off" then you have more general problems than WSL. Windows Update to latest (currently 2004 aka 20H1) can sometimes solve many ills by virtue of reinstalling all the things. Tangentially similar #3689.

**Comment by licanhua** [comment]
Are you using x86 OS? only AMD64 or ARM64 supports WSL

**Comment by roaringsundew40** [comment]
@therealkenc  I have a Dell Computer and since my computer is very old it is not updating. According to Dell my computer is not compatible for any version then what I have right now. 

@licanhua I have a X64 OS.

**Comment by therealkenc** [comment]
For the giggles, try running the following to see if you can install any optional components (nvm WSL). From an elevated Powershell:
```
PS C:\> Get-WindowsOptionalFeature -Online
```

If it returns anything (but presumably not `Microsoft-Windows-Subsystem-Linux`) try enabling something harmless like `TelnetClient`.

I found some [lore](https://www.winhelponline.com/blog/fix-turn-windows-features-on-or-off-is-blank-in-windows-10/) on the web that might or might not help. It comes with no endorsement.

Another popular path forward, assuming your box is disposable, is to disregard Dell, install Windows from an [image](https://www.microsoft.com/en-ca/software-download/windows10), and hope for the best. Sometimes it works out. YRMV.

**Comment by roaringsundew40** [comment]
`PS C:\Windows\system32>  Get-WindowsOptionalFeature -Online
Get-WindowsOptionalFeature : Get-WindowsOptionalFeature failed. Error code = 0x800f0831
At line:1 char:2
+  Get-WindowsOptionalFeature -Online
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-WindowsOptionalFeature], COMException
    + FullyQualifiedErrorId : Microsoft.Dism.Commands.GetWindowsOptionalFeatureCommand
`

You may be right... I might not be able to use Get-WindowsOptionalFeature.

**Comment by therealkenc** [comment]
Yes, sadly there isn't a WSL actionable here (anymore than there is a `TelnetClient` actionable). Do hope you can get your old box whole again.

**Comment by roaringsundew40** [comment]
Ok Thanks for your help
